src/date.ts

```diff
@@ -0,0 +1,151 @@
+import 'globalize/dist/globalize/date';
+import 'globalize/dist/globalize/relative-time';
+import getGlobalize, { normalizeOptions } from './util/globalize';
+
+export type DateFormatterOptions = null | {
+	/**
+	 * String value indicating a skeleton, eg. { skeleton: "GyMMMd" }.
+	 * Skeleton provides a more flexible formatting mechanism than the predefined list full, long, medium, or short represented by date, time, or datetime.
+	 * Instead, they are an open-ended list of patterns containing only date field information, and in a canonical order.
+	 */
+	skeleton?: string;
+
+	/**
+	 * One of the following String values: full, long, medium, or short, eg. { date: "full" }.
+	 */
+	date?: DateLength;
+
+	/**
+	 * One of the following String values: full, long, medium, or short, eg. { time: "full" }.
+	 */
+	time?: DateLength;
+
+	/**
+	 * One of the following String values: full, long, medium, or short, eg. { datetime: "full" }
+	 */
+	datetime?: DateLength;
+}
+
+export type DateLength = 'short' | 'medium' | 'long' | 'full';
```

Comment:
In any case, since this particular style is a suggestion, let me know if you still think it would be beneficial to arrange the interface/type declarations in this manner.